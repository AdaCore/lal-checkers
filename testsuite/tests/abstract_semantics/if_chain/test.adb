procedure Ex1 is
   x : Integer;
   C1, C2, C3 : Boolean;
begin
   if C1 then
      x := 1;
   elsif C2 then
      x := 2;
   elsif C3 then
      x := 3;
   else
      x := 4;
   end if;
end Ex1;
