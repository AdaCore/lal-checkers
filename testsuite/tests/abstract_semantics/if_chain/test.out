{
  "always": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "C1": "{True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[1, 1]"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign5", 
          "assume1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[2, 2]"
        }
      }
    ], 
    "assign6": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assume1", 
          "assume3", 
          "assume4", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[3, 3]"
        }
      }
    ], 
    "assign7": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume5", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[4, 4]"
        }
      }
    ], 
    "assume0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "C1": "{True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "assume4", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "assume5", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "split_join0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume4", 
          "assume5", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False, True}", 
          "x": "[3, 4]"
        }
      }
    ], 
    "split_join1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign5", 
          "assign6", 
          "assign7", 
          "assume1", 
          "assume2", 
          "assume3", 
          "assume4", 
          "assume5", 
          "split_join0", 
          "split_join1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[2, 4]"
        }
      }
    ], 
    "split_join2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assign5", 
          "assign6", 
          "assign7", 
          "assume0", 
          "assume1", 
          "assume2", 
          "assume3", 
          "assume4", 
          "assume5", 
          "split_join0", 
          "split_join1", 
          "split_join2", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[1, 4]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ]
  }, 
  "le_t_eq_v": {
    "assign0": [
      {
        "trace:": [
          "assign0", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assign4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "C1": "{True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[1, 1]"
        }
      }
    ], 
    "assign5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign5", 
          "assume1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[2, 2]"
        }
      }
    ], 
    "assign6": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assume1", 
          "assume3", 
          "assume4", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[3, 3]"
        }
      }
    ], 
    "assign7": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume5", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[4, 4]"
        }
      }
    ], 
    "assume0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume0", 
          "start0"
        ], 
        "values": {
          "C1": "{True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume3": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume4": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "assume4", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "assume5": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assume1", 
          "assume3", 
          "assume5", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ], 
    "split_join0": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assume1", 
          "assume3", 
          "assume4", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[3, 3]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume5", 
          "split_join0", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[4, 4]"
        }
      }
    ], 
    "split_join1": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign5", 
          "assume1", 
          "assume2", 
          "split_join1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[2, 2]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assume1", 
          "assume3", 
          "assume4", 
          "split_join0", 
          "split_join1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[3, 3]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume5", 
          "split_join0", 
          "split_join1", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[4, 4]"
        }
      }
    ], 
    "split_join2": [
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign4", 
          "assume0", 
          "split_join2", 
          "start0"
        ], 
        "values": {
          "C1": "{True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[1, 1]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign5", 
          "assume1", 
          "assume2", 
          "split_join1", 
          "split_join2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{True}", 
          "C3": "{False, True}", 
          "x": "[2, 2]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign6", 
          "assume1", 
          "assume3", 
          "assume4", 
          "split_join0", 
          "split_join1", 
          "split_join2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{True}", 
          "x": "[3, 3]"
        }
      }, 
      {
        "trace:": [
          "assign0", 
          "assign1", 
          "assign2", 
          "assign3", 
          "assign7", 
          "assume1", 
          "assume3", 
          "assume5", 
          "split_join0", 
          "split_join1", 
          "split_join2", 
          "start0"
        ], 
        "values": {
          "C1": "{False}", 
          "C2": "{False}", 
          "C3": "{False}", 
          "x": "[4, 4]"
        }
      }
    ], 
    "start0": [
      {
        "trace:": [
          "start0"
        ], 
        "values": {
          "C1": "{False, True}", 
          "C2": "{False, True}", 
          "C3": "{False, True}", 
          "x": "[-2147483648, 2147483647]"
        }
      }
    ]
  }
}
